varats/varats/data/databases/feature_perf_precision_database.py

```diff
@@ -0,0 +1,932 @@
+
+    def __init__(self) -> None:
+        super().__init__(
+            "eBPFTrace", fpp.EbpfTraceTEFProfileRunner,
+            fpp.TEFProfileOverheadRunner, fpp.MPRTEFAggregate
+        )
+
+    def is_regression(
+        self, report_path: ReportFilepath, patch_name: str
+    ) -> bool:
+        """Checks if there was a regression between the old an new data."""
+        multi_report = MultiPatchReport(
+            report_path.full_path(), TEFReportAggregate
+        )
+
+        old_acc_pim: tp.DefaultDict[str, tp.List[int]] = defaultdict(list)
+        for old_tef_report in multi_report.get_baseline_report().reports():
+            pim = get_feature_performance_from_tef_report(old_tef_report)
+            for feature, value in pim.items():
+                old_acc_pim[feature].append(value)
+
+        new_acc_pim: tp.DefaultDict[str, tp.List[int]] = defaultdict(list)
+        opt_mr = multi_report.get_report_for_patch(patch_name)
+        if not opt_mr:
+            raise NotImplementedError()
+
+        for new_tef_report in opt_mr.reports():
+            pim = get_feature_performance_from_tef_report(new_tef_report)
+            for feature, value in pim.items():
+                new_acc_pim[feature].append(value)
+
+        return self.pim_regression_check(old_acc_pim, new_acc_pim)
+
+
+class Baseline(Profiler):
+    """Profiler mapper implementation for the black-box baseline."""
+
+    def __init__(self) -> None:
+        super().__init__(
+            "Base", fpp.BlackBoxBaselineRunner, fpp.BlackBoxOverheadBaseline,
+            TimeReportAggregate
+        )
+
+    def is_regression(
+        self, report_path: ReportFilepath, patch_name: str
+    ) -> bool:
+        time_reports = load_mpr_time_report_aggregate(report_path)
+
+        old_time = time_reports.get_baseline_report()
+        new_time = time_reports.get_report_for_patch(patch_name)
+        if not new_time:
+            raise LookupError(f"Missing new time report in file {report_path}")
+
+        # Cut off regressions smaller than 100ms
+        req_diff = self.absolute_cut_off / 1000
+        if np.mean(old_time.measurements_wall_clock_time
+                  ) == np.mean(new_time.measurements_wall_clock_time):
+            return False
+
+        if abs(
+            np.mean(old_time.measurements_wall_clock_time) -
+            np.mean(new_time.measurements_wall_clock_time)
+        ) < req_diff:
+            return False
+
+        return self.default_regression_check(
+            old_time.measurements_wall_clock_time,
+            new_time.measurements_wall_clock_time
+        )
+
+
+def get_patch_names(case_study: CaseStudy) -> tp.List[str]:
+    """Looks up all patch names from the given case study."""
+    report_files = get_processed_revisions_files(
+        case_study.project_name,
+        fpp.BlackBoxBaselineRunner,
+        fpp.MPRTimeReportAggregate,
+        get_case_study_file_name_filter(case_study),
+        config_id=0
+    )
```

Comment:
Wouldn't it be better to also make the `config_id` a parameter?
```suggestion
def get_patch_names(case_study: CaseStudy, config_id: int = 0) -> tp.List[str]:
    """Looks up all patch names from the given case study."""
    report_files = get_processed_revisions_files(
        case_study.project_name,
        fpp.BlackBoxBaselineRunner,
        fpp.MPRTimeReportAggregate,
        get_case_study_file_name_filter(case_study),
        config_id=config_id
    )
```
There might be scenarios in which you have different patches for different configs (At least this was the case in my thesis)

Reply:
Hmm, I'm not sure if that can/should happen here in this case. In generell, I would agree with you, however here we fetch the patch names for our regression patches that we compare in the experiment. If the returned names would differ, the experiment setup would have a problem. So, I'm not sure if we should make the config_id a variable part here.